askama_derive/src/input.rs

```diff
@@ -104,15 +104,15 @@ impl TemplateInput<'_> {
         &self,
         map: &mut HashMap<PathBuf, Parsed>,
     ) -> Result<(), CompileError> {
-        let source = match &self.source {
-            Source::Source(s) => s.clone(),
-            Source::Path(_) => get_template_source(&self.path)?,
+        let (source, source_path) = match &self.source {
+            Source::Source(s) => (s.clone(), None),
+            Source::Path(_) => (get_template_source(&self.path)?, Some(self.path.clone())),
```

Comment:
Do we actually need to `clone()` here? Why?

Reply:
Unless I missed something, we can't pass `self.path` as value because it's part of `self`. And if we pass as reference, it'll create lifetime issues below in the `add_to_check` closure. Do you see a way to go around that?

Reply:
Something I thought about: we could use `Rc` to prevent having all these clones. What do you think?